Migrate useOrganization from Sanity to MDX (#332)
clerkinc/clerk-docs: docs/references/react/use-organization.mdx

---
title: useOrganization()
description: Access attributes of the currently active organization.
---

# `useOrganization()`

> The `useOrganization()` hook gives you access to the current active organization attributes.
> 
> ```jsx
> type useOrganizationParams = {
>   invitationList: {
>     // Pagination params
>     limit?: number;
>     offset?: number;
>     },
>   membershipRequests: true
>     | (commonPaginationParams & {
>         status?: OrganizationInvitationStatus; // e.g. "pending"
>       });
>   memberships: true
>     | (commonPaginationParams & {
>         role?: MembershipRole[]; // e.g. ["basic_member", "admin"]
>       });
>   domains: true
>     | (commonPaginationParams & {
>         enrollmentMode?: OrganizationEnrollmentMode; // e.g. "automatic_invitation"
>       });
> };
> 
> type commonPaginationParams = {
>   initialPage?: number;
>   pageSize?: number;
>   infinite?: boolean;
>   keepPreviousData?: boolean;
> }

Author: So we have a /types context. I'm thinking we can define types here, such as `useOrganizationParams`, `commonPaginationParams`, `OrganizationInvitationStatus`, `MembershipRole`, and `OrganizationEnrollmentMode`.

```suggestion
The `useOrganization()` hook gives you access to the current active organization attributes. This hook accepts [`useOrganizationParams`](/docs/refereces/javascript/types/use-organization-params).
```jsx
```

});
// invitationList is retrieved and auto-updating as expected.
```

In addition, we support a paginated response with page fetching helper functions for `memberships`, `membershipRequests`, and `domains`. See [usage](#usage) below for an example.

## Usage

<Callout type="warning">
  Make sure you've followed the installation guide for [Clerk React](/docs/references/react/overview) before running the snippets below.
</Callout>

In the following example, `useOrganization()` is used to map over the `memberships` of the current active organization and present membership attributes. In addition, buttons support navigating through the paginated response.

<Callout type="info">
  Note that your component must be a descendant of [`<ClerkProvider/>`](/docs/components/clerk-provider).
</Callout>

```jsx
import { useOrganization } from "@clerk/nextjs";

export default function MemberList() {
  const { memberships } = useOrganization({
    memberships: {
      infinite: true,
      keepPreviousData: true,
    },
  });

  if (!memberships) {
    // loading state
    return null;
  }

  return (
    <div>
      <h2>Organization members</h2>
      <ul>

>         {memberships.map((membership) => (

Reviewer: ```suggestion
        {memberships.data?.map((membership) => (
```

>         {memberships.data.map((membership) => (

Reviewer: data may be undefined here
```suggestion
        {memberships.data?.map((membership) => (
```
